### DISCUSSION TITLE
What is Collect metric which is coming after vitest run

### DISCUSSION THREAD

**Discussion by DevathiMahesh** [vitest-dev/vitest#2348] [Q&A] [answered]
Hi Folks,
When i run test cases using vitest, i am getting 4 metrics (Transform, Setup, Collect, tests) and each one of them takes some seconds to complete their functionality. I know **transform** is for performing transformations (Typescript etc...0, **Setup** for initial setup while running test cases, **tests** for actual time tests are running. What exactly **Collect** means ?? 
My problem is collect taking more time bcoz of that the total running time is increasing. Is there a way to reduce time taken by **collect**  and if so please reply to this thread.... 
Thanks in advance,

**Comment by xel1045** [comment, marked as answer]
This questions was already answered here: https://github.com/vitest-dev/vitest/discussions/1770

**Reply by DevathiMahesh** [reply]
@xel1045  My question is there anyway to optimise the collect attribute so that overall running time of the tests comes down???

**Reply by xel1045** [reply]
Unfortunately no, I have the same problem which brings me here! :P

**Reply by DevathiMahesh** [reply]
Ummm. It could have been better if there is a some way to optimise this metric also. Anyway closing this thread.

**Reply by andyroschy** [reply]
@DevathiMahesh A little late to the party but for posterity.

It is possible to improve the collect time by optimizing the number of dependencies (direct and indirect) that need to be resolved by any given test. 

Avoiding barrel files can help, since importing any single value from a barrel file implies resolving all the imports on the barrel file.
You can also try analyzing your dependency graph with something like dependency-cruiser to see if you are unnecessarily resolving dependencies that are not needed for the components you are testing.
